src/dnf-plugins/product-id/product-id.c

```diff
@@ -0,0 +1,640 @@
+        if (productIdFound) {
+            gchar *productId = g_strdup(outname->str);
+            g_string_prepend(outname, PRODUCT_CERT_DIR);
+            g_string_append(outname, ".pem");
+            FILE *fileOutput = fopen(outname->str, "w+");
+            if (fileOutput != NULL) {
+                debug("Content of certificate written to: %s", outname->str);
+                fprintf(fileOutput, "%s", pemOutput->str);
+                fclose(fileOutput);
+
+                addRepoId(productDb, productId, dnf_repo_get_id(repoProductId->repo));
+                ret = 1;
+            } else {
+                error("Unable write to file with certificate file :%s", outname->str);
+            }
+            g_free(productId);
+        }
+
+        out:
+        g_string_free(outname, TRUE);
+        g_string_free(pemOutput, TRUE);
+    }
+
+    if(input != NULL) {
+        gzclose(input);
+    } else {
+        debug("Unable to open compressed product certificate");
+    }
+
+    return ret;
+}
+
+/**
+ * Look at the PEM of a certificate and figure out what is ID of the product.
+ *
+ * @param certContent String containing content of product certificate
+ * @param result String containing ID of of product certificate
+ * @return
+ */
+int findProductId(GString *certContent, GString *result) {
+    int ret_val = 1;
+    BIO *bio = BIO_new_mem_buf(certContent->str, (int) certContent->len);
+    if (bio == NULL) {
+        debug("Unable to create buffer for content of certificate: %s",
+                ERR_error_string(ERR_get_error(), NULL));
+        return -1;
+    }
+
+    X509 *x509 = PEM_read_bio_X509(bio, NULL, NULL, NULL);
+    BIO_free(bio);
+    bio = NULL;
+
+    if (x509 == NULL) {
+        debug("Failed to read content of certificate from buffer to X509 structure: %s",
+                ERR_error_string(ERR_get_error(), NULL));
+        return -1;
+    }
+
+    int exts = X509_get_ext_count(x509);
+    gboolean redhat_oid_found = FALSE;
+    for (int i = 0; i < exts; i++) {
+        char oid[MAX_BUFF];
+        X509_EXTENSION *ext = X509_get_ext(x509, i);
+        if (ext == NULL) {
+            debug("Failed to get extension of X509 structure: %s",
+                  ERR_error_string(ERR_get_error(), NULL));
+            ret_val = -1;
+            break;
+        }
+        OBJ_obj2txt(oid, MAX_BUFF, X509_EXTENSION_get_object(ext), 1);
+
+        if (strncmp(REDHAT_PRODUCT_OID, oid, strlen(REDHAT_PRODUCT_OID)) == 0) {
+            redhat_oid_found = TRUE;
+            gchar **components = g_strsplit(oid, ".", -1);
+            int comp_id=0;
+            // Because g_strsplit() returns array of NULL terminated pointers,
+            // then we have to make sure that array is long enough to contain
+            // required ID
+            while(components[comp_id] != NULL) {
+                comp_id++;
+            }
+            debug("Number of OID components: %d", comp_id);
```

Comment:
Delete this.  Not very helpful as debugging info.  More helpful would be to print the entire OID.